src/components/computerVsHumanGameComponent/computerVsHumanGameComponent.js

```diff
@@ -30,27 +30,50 @@ class ComputerVsHumanGameComponent extends Component {
       let next_player = this.state.computer.symbol;
       let current_player = this.state.human;
       let marks = this.state.board.marks();
-      this.setState({
-        gameStatus: await PhoenixApi.getStatus(marks, next_player, current_player),
-      });
 
-      if(this.state.gameStatus === statuses.IN_PROGRESS){
-        this.setState({
-          gameStatus: statuses.COMPUTER_THINKING
-        });
-        setTimeout( () => {this.computerMove();}, 1000);
-      }
+      await this.updateGameStatus(marks, next_player, current_player);
+      this.checkForWinner(marks);
+      this.checkForComputerTurn();
     }
 
     computerMove = async () => {
       let board = this.state.board;
       let human = this.state.human;
       await this.state.computer.makeMove(board, human);
+
+      await this.updateGameStatus(board.marks(), human, this.state.computer.symbol);
+      this.checkForWinner(board.marks());
+    }
+
+    updateGameStatus = async (marks, next_player, current_player) => {
       this.setState({
-        gameStatus: await PhoenixApi.getStatus(board.marks(), human, this.state.computer.symbol),
+        gameStatus: await this.state.api.getStatus(marks, next_player, current_player)
       });
     }
 
+    checkForComputerTurn = () => {
+      if(this.state.gameStatus === Statuses.IN_PROGRESS){
+        this.setState({
+          gameStatus: Statuses.COMPUTER_THINKING
+        });
+        setTimeout(() => {this.computerMove()}, 1000);
```

Comment:
Why is the `setTimeout` needed here? It is not wrong, per se, just a bit weird to see the computer move behind the timeout.

Reply:
Initially the computer was placing a marker very quickly.  So we placed `setTimeout` here to extend the time the computer makes a turn and enhance the users' experience.